### ISSUE TITLE
Some systems failed to build for Quicklisp dist

### ISSUE THREAD

**Issue by quicklisp** [digikar99/numericals#4]
Building with SBCL 2.3.1.122-1455d60f6 / ASDF 3.3.5 for quicklisp dist creation.

Trying to build commit id a8995ea9e976810b74e6910986a66ae16a354f47

*dense-numericals* fails to build with the following error:

```
; caught ERROR:
;   (during macroexpansion of (DEFPOLYMORPH (NU:COPY :MORE-OPTIMAL-TYPE-LIST ...) ...))
;   Can't match (((OUT (ARRAY <TYPE>)))) with ((LIST* (LIST POLYMORPHIC-FUNCTIONS::PARAMETER TYPE) POLYMORPHIC-FUNCTIONS::DEFAULT POLYMORPHIC-FUNCTIONS::SUPPLIED-P)).
...
; caught ERROR:
;   (during macroexpansion of (DEFPOLYMORPH (NU:COPY :SUBOPTIMAL-NOTE ...) ...))
;   Can't match (((OUT NULL))) with ((LIST* (LIST POLYMORPHIC-FUNCTIONS::PARAMETER TYPE) POLYMORPHIC-FUNCTIONS::DEFAULT POLYMORPHIC-FUNCTIONS::SUPPLIED-P)).
...
; caught ERROR:
;   (during macroexpansion of (DEFPOLYMORPH NU:COPY ...))
;   Can't match (((OUT (SIMPLE-ARRAY <TYPE>)))) with ((LIST* (LIST POLYMORPHIC-FUNCTIONS::PARAMETER TYPE) POLYMORPHIC-FUNCTIONS::DEFAULT POLYMORPHIC-FUNCTIONS::SUPPLIED-P)).
...
; caught ERROR:
;   (during macroexpansion of (DEFPOLYMORPH (NU:COPY :MORE-OPTIMAL-TYPE-LIST ...) ...))
;   Can't match (((OUT (ARRAY SINGLE-FLOAT)))) with ((LIST* (LIST POLYMORPHIC-FUNCTIONS::PARAMETER TYPE) POLYMORPHIC-FUNCTIONS::DEFAULT POLYMORPHIC-FUNCTIONS::SUPPLIED-P)).
...
; caught ERROR:
;   (during macroexpansion of (DEFPOLYMORPH NU:COPY ...))
;   Can't match (((OUT (SIMPLE-ARRAY SINGLE-FLOAT)))) with ((LIST* (LIST POLYMORPHIC-FUNCTIONS::PARAMETER TYPE) POLYMORPHIC-FUNCTIONS::DEFAULT POLYMORPHIC-FUNCTIONS::SUPPLIED-P)).
...
; caught ERROR:
;   (during macroexpansion of (DEFPOLYMORPH (NU:COPY :MORE-OPTIMAL-TYPE-LIST ...) ...))
;   Can't match (((OUT (ARRAY DOUBLE-FLOAT)))) with ((LIST* (LIST POLYMORPHIC-FUNCTIONS::PARAMETER TYPE) POLYMORPHIC-FUNCTIONS::DEFAULT POLYMORPHIC-FUNCTIONS::SUPPLIED-P)).
...
; caught ERROR:
;   (during macroexpansion of (DEFPOLYMORPH NU:COPY ...))
;   Can't match (((OUT (SIMPLE-ARRAY DOUBLE-FLOAT)))) with ((LIST* (LIST POLYMORPHIC-FUNCTIONS::PARAMETER TYPE) POLYMORPHIC-FUNCTIONS::DEFAULT POLYMORPHIC-FUNCTIONS::SUPPLIED-P)).
...
Unhandled UIOP/LISP-BUILD:COMPILE-FILE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING {1040610113}>: COMPILE-FILE-ERROR while compiling #<CL-SOURCE-FILE "dense-numericals" "copy-coerce">

```

*numericals* fails to build with the following error:

```
; caught ERROR:
;   (during macroexpansion of (DEFPOLYMORPH (NU:ASARRAY :INLINE ...) ...))
;   Can't match (((TYPE (EQL :AUTO)))) with ((LIST* (LIST POLYMORPHIC-FUNCTIONS::PARAMETER TYPE) POLYMORPHIC-FUNCTIONS::DEFAULT POLYMORPHIC-FUNCTIONS::SUPPLIED-P)).
...
Unhandled UIOP/LISP-BUILD:COMPILE-FILE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING {1001728003}>: COMPILE-FILE-ERROR while compiling #<CL-SOURCE-FILE "numericals" "primitives">

```

*numericals/benchmarks* fails to build because of a failure in  _numericals_.

[Full log here](http://report.quicklisp.org/2023-04-17/failure-report/numericals.html)

**Comment by digikar99** [comment]
Okay, I have published the [tag v0.4.0 for polymorphic-functions](https://github.com/digikar99/polymorphic-functions/releases/tag/v0.4.0) which should resolve this issue.

**Comment by digikar99** [comment]
I'm closing this since another build attempt is being made.
